docs/DeveloperGuide.md

```diff
@@ -1,35 +1,298 @@
+
+Step 3. The user has received an update about an extension to the deadline for the task, and decides to change
+the deadline by executing the `edit 1 -due 25/03/2023 15:00` command. The `edit` will call `TaskList#setDeadline()`,
+which updates the value of deadline for the Task item saved at id 1 to the new updated deadline.
+
+The following sequence diagram shows how the edit operations works:
+
+![EditDeadlineCommandSequence](images/EditDeadlineCommandSequence.png)
+
+### List tasks sorted by deadline feature
+
+This ListTasksCommand extends NUS To-Do List with an automatic sorting feature that sorts all tasks in an ascending
+deadline order and displays the To-Do List to users. It is facilitated by the TaskList, Command class, and UI class.
+It implements the `TaskList#toString()` and  `ui#printTaskList()`operation.
+
+Given below is an example usage scenario and how the ListTasksCommand mechanism will behave at each step.
+
+Step 1. The user launches the application for the first time. There are no existing tasks read by the program.
+
+Step 2. The user executes `add survey -due 20/03/2023 12:00` and `add survey -due 21/03/2023 12:00` command to add 2
+tasks to the To-Do List. The `add` command calls `TaskList#addTask()` once for each task, which causes 2 new Tasks to
+be added to the existing TaskList.
+
+Step 3. The user wants to view the entire list of deadlines that he/she has added. The user can do this by using the
+command `list` into the terminal. By doing so,`TaskList#toString()` will be executed, which will sort
+the list in an ascending deadline order, where the deadline that is closest to date will be at the top and the deadline
+furthest to date will be at the bottom. Next, `ui#printTaskList()` will be executed, which will display the list of
+deadlines to the user in the terminal.
+
+The following sequence diagram shows how the list operation works:
+
+![ListTasksCommandSequence](images/ListTasksCommandSequence.png)
+
+### Storage feature
+
+The storage feature is facilitated by the `Storage` class.
+
+The Storage class implements the following operations:
+- `Storage#saveData(TaskList)` --- Saves the current task list.
+- `Storage#loadData()` --- Loads a task list from the previously saved file, if there is one.
+
+Given below is an example usage scenario and how the Storage mechanism behaves at each step.
+
+Step 1. The user launches the application (not for the first time). The program loads the previously saved task list 
+data as there is a saved file `'./data.txt'` that Storage can find.
+
+![Step 1](images/StorageSequenceDiagramStep1.png)
+
+Step 2. The user executes `list` command. The `list` command calls `Ui#printTaskList()` which lists all tasks in the 
+task list. `ToDoListManager` calls `storage#saveData()`, so the task list is saved into `'./data.txt'`.
+
+![Step2](images/StorageSequenceDiagramStep2-Step_2.png)
```

Comment:
I think this diagram could be broken up into two pictures and/or place more focus on the methods more closely related to `list` command, as after a few diagrams, it may be redundant to showcase the user and parser.

Reply:
Also, I am slightly confused at Step 7 onwards in this diagram. Should there be return values by `TaskList`?